src/proxy/sessions/session.rs

```diff
@@ -100,36 +111,44 @@ impl Packet {
     pub fn contents(&self) -> &[u8] {
         &self.contents
     }
+
+    /// stop and record the stored [HistogramTimer]
+    pub fn stop_and_record(self) {
+        self.timer.stop_and_record();
+    }
+}
+
+pub struct SessionArgs {
```

Comment:
I did this because Clippy complained there were too many arguments going to `Session::new(....)` -- I'm not sure if this actually improves things? It seems to just push the problem down a level?

I'd love to hear if there is a better way of doing this?

Reply:
Instead of `SessionArgs`, and passing that to `Session::new` I would instead make `new` a method on `SessionArgs` similar to the pseudo-actor run pattern I've been using elsewhere.


```rust
SessionArgs {
    metrics,
    proxy_metrics,
    filter_manager,
    from,
    dest,
    sender,
    ttl,
}.new(logger)
```